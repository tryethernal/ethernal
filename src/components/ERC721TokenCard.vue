<template>
    <v-card>
        <router-link v-if="imageData || owner" :to="`/token/${contractAddress}/${tokenId}`">
            <div :style="`background-color: ${backgroundColor ? '#' + backgroundColor : ''};`" rounded="lg" elevation="2" v-html="imageData"></div>
        </router-link>
        <v-skeleton-loader v-else type="image"></v-skeleton-loader>
        <v-card-text>
            <router-link v-if="name" :to="`/token/${contractAddress}/${tokenId}`">
                <b>{{ name }}</b>
            </router-link>
            <v-skeleton-loader v-else type="heading"></v-skeleton-loader>
            <br/>
            <template v-if="owner">
                Owned by <Hash-Link :type="'address'" :hash="owner" :xsHash="true" />
            </template>
            <v-skeleton-loader v-else type="text"></v-skeleton-loader>
        </v-card-text>
    </v-card>
</template>

<script>
import HashLink from './HashLink';

export default {
    name: 'ERC721TokenCard',
    props: ['owner', 'tokenId', 'imageData', 'name', 'contractAddress', 'backgroundColor'],
    components: {
        HashLink
    },
}
</script>
