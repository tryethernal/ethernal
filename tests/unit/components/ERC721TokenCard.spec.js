import MockHelper from '../MockHelper';
import flushPromises from 'flush-promises'

import ERC721TokenCard from '@/components/ERC721TokenCard.vue';

const helper = new MockHelper();

describe('ERC721TokenCard.vue', () => {
    beforeEach(() => jest.clearAllMocks());

    it('Should display an erc721 token card', async () => {
        const wrapper = helper.mountFn(ERC721TokenCard, {
            propsData: {
                owner: '0x123',
                tokenId: '0',
                imageData: '<img src="http://image" />',
                name: 'Token #1',
                contractAddress: '0xabc',
                backgroundColor: '#1234'
            },
            stubs: ['Hash-Link']
        });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
    });
});
