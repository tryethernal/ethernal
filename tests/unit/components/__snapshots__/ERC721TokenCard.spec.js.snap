// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ERC721TokenCard.vue Should display an erc721 token card 1`] = `
<div class="v-card v-sheet theme--light"><a href="#/token/0xabc/0" class="">
    <div rounded="lg" elevation="2"><img src="http://image"></div>
  </a>
  <div class="v-card__text"><a href="#/token/0xabc/0" class=""><b>Token #1</b></a> <br>
    Owned by <hash-link-stub xshash="true" hash="0x123" type="address"></hash-link-stub>
  </div>
</div>
`;
